Documentation/design/initphase.md

```diff
@@ -0,0 +1,44 @@
+# Init Phase
+
+## Goals
+
+1. Generates `InstallConfig` by prompting user for required parameters, and populating the rest of configuration with default values.
+2. Make useful decisions on default values for most options.
+3. Init phase should never prompt the user for more than 10 options unless there is a very strong reason.
+
+## Non Goals
+
+1. Supporting all the configuration option through init is not a goal. Init is just a helper for humans.
+
+## Overview
+
+The output of the `init` command is a config file that describes a default cluster for the given provider, and is valid/immediately usable.
+
+The `init` step is not required, this is merely a helper to generate a valid `InstallConfig`. A user could use examples from documentation, roll their own by hand, or re-use existing configs from source control.
+
+## Detailed Design
+
+### Prompts on CLI vs flags vs Environment variables
+
+1. Flags don’t work well here because we want the fast-path to be single command.
+2. Init is only valuable for humans and environment variables are similarly tedious.
+
+Init phase should use prompts on CLI for top-level configurations that are required to generate InstallConfig.
```

Comment:
I think you want backticks for `InstallConfig` here to match the rest of the file.